File: codes/cpp/encapsulation/person.cpp
#include "person.h"
#include <iostream>

int Person::nbr = 0;

Person::Person(std::string name)
{
	this->name = name;
	num = (Person::nbr++);
	t = "person";
	luckyNumber = 0;
}

void Person::info()
{
	std::cout << "My name: " << name << "\n";
	std::cout << "My number: " << num << "\n";
	std::cout << "I am  a: " << t << "\n";
	std::cout << "My lucky number is: " << luckyNumber << "\n";
	std::cout << "--------------------------\n";
}

void Person::copy(Person other)
{
	name = other.name;
	t = other.t;
}

File: codes/php/encapsulation/person.php
<?php
class Person {

    public $luckyNumber = 0;
    var $luckyNumber2 = 0;//public by default
    protected $t;
    private $name;
    private $num;
    private static $nbr = 0;

    public function __construct( $name) {
        $this->name = $name;
        $this->num = (self::$nbr++);
        $this->t = "person";
    }

    public function info() {
        echo "My name: $this->name\n";
        echo "My number: $this->num\n";
        echo "I am a: $this->t\n";
        echo "My lucky number is: $this->luckyNumber\n";
        echo "My lucky number 2 is: $this->luckyNumber2\n";
        echo "--------------------------\n";
    }

    public function copy($other) {
        $this->name =$other->name;
        $this->t = $other->t;
    }

}

?>
